### PR TITLE
scopes: OAuth2 scopes definition and example app improvement

### PR DESCRIPTION
**Pull request by hachreak**

### CODE REVIEW

**Comment by hachreak** [comment]
depends on https://github.com/inveniosoftware/invenio-oauth2server/pull/36

ping @lnielsen

File: invenio_deposit/scopes.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+class Write(DepositScope):
+    """Scope to upload a deposit."""
+
+    scope_id = "deposit:write"
+
+    def __init__(self, *args, **kwargs):
+        """Define the scope."""
+        super(Write, self).__init__(
+            help_text="Allow upload (but not publishing).",
```
Comment (jirikuncar) [inline invenio_deposit/scopes.py]: Please use single quotes and mark help text for i18n with `_` function.

File: invenio_deposit/scopes.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    """Deposit scope."""
+
+    def __init__(self, *args, **kwargs):
+        """Define the scope."""
+        super(DepositScope, self).__init__(
+            id_=self.scope_id, group="deposit", *args, **kwargs
+        )
+
+
+class Write(DepositScope):
```
Comment (jirikuncar) [inline invenio_deposit/scopes.py]:
```python
class DepositScope(Scope):
    """Deposit scope."""

    def __init__(self, id_, *args, **kwargs):
        """Define the scope."""
        super(DepositScope, self).__init__(
            id_='deposit:{0}'.format(id_),
            group='deposit', *args, **kwargs
        )

write_scope = DepositScope('write', help_text=_('Allow upload (but not publishing).'))
actions_scope = DepositScope('write', help_text=_('Allow publishing of uploads.'))
```

File: setup.py [hunk 3]
```diff
@@ -128,6 +128,10 @@
         'invenio_search.mappings': [
             'deposits = invenio_deposit.mappings',
         ],
+        'invenio_oauth2server.scopes': [
+            'deposit_write = invenio_deposit.scopes:Write',
```
Comment (jirikuncar) [inline setup.py]:
```python
'deposit_write = invenio_deposit.scopes:deposit_write',
'deposit_actions = invenio_deposit.scopes:deposit_actions',
```

File: invenio_deposit/views/rest.py [hunk 4]
```diff
@@ -200,6 +201,8 @@ def get(self, pid, record):
         """Get deposit/depositions/:id/files."""
         return self.make_response(record.files)
 
+    @require_api_auth()
+    @require_oauth_scopes("deposit:write")
```
Comment (jirikuncar) [inline invenio_deposit/views/rest.py]: `@require_oauth_scopes(write_scope.id)`

File: requirements-devel.txt [hunk 5]
```diff
@@ -31,3 +31,4 @@
 -e git+git://github.com/inveniosoftware/invenio-search-ui.git#egg=invenio-search-ui
 -e git+git://github.com/inveniosoftware/invenio-search.git#egg=invenio-search
 -e git+git://github.com/inveniosoftware/invenio-files-rest.git#egg=invenio-files-rest
+-e git+git://github.com/inveniosoftware/invenio-oauth2server.git#egg=invenio-oauth2server
```
Comment (jirikuncar) [inline requirements-devel.txt]: Please keep the file sorted.

**Comment by hachreak** [comment]
ping @lnielsen

File: examples/requirements.txt [hunk 6]
```diff
@@ -34,3 +34,6 @@ invenio-records-ui>=1.0.0a4
 invenio-records>=1.0.0a9
 invenio-search>=1.0.0a3
 invenio-theme>=1.0.0a7
+invenio-admin>=1.0.0a3
```
Comment (jirikuncar) [inline examples/requirements.txt]: please keep the requirements sorted

File: setup.py [hunk 7]
```diff
@@ -75,6 +75,7 @@
     'invenio-records>=1.0.0a15',
     'invenio-search-ui>=1.0.0a4',
     'invenio-search>=1.0.0a7',
+    'invenio-oauth2server>=1.0.0a1',
```
Comment (jirikuncar) [inline setup.py]: please keep the list of requirements sorted

File: tests/conftest.py [hunk 8]
```diff
@@ -95,6 +117,58 @@ def app(request):
...
+        # create resource_owner -> client_1
+        client_ = Client(
+            client_id='client_test_u1c1',
+            client_secret='client_test_u1c1',
+            name='client_test_u1c1',
+            description='',
+            is_confidential=False,
+            user=users[0],
+            _redirect_uris='',
+            _default_scopes=""
```
Comment (jirikuncar) [inline tests/conftest.py]: `_default_scopes='',`

**Comment by hachreak** [comment]
@lnielsen the PR is ready!
We need invenio-oauth2server 1.0.0a2 is released: see https://github.com/inveniosoftware/invenio-oauth2server/pull/37
can you review it? thanks :)

**Pull request merged**